src/components/camera/server.cpp

```diff
@@ -0,0 +1,162 @@
+#include <components/camera/server.hpp>
+#include <components/camera/camera.hpp>
+#include <rpc/server.hpp>
+#include <gen/google/api/http.pb.h>
+
+::grpc::Status CameraServer::DoCommand(
+    ::grpc::ServerContext* context,
+    const ::viam::common::v1::DoCommandRequest* request,
+    ::viam::common::v1::DoCommandResponse* response) {
+
+    std::shared_ptr<ResourceBase> rb = sub_svc->resource(request->name());
+    if (rb == nullptr) {
+        return grpc::Status(grpc::UNKNOWN, "resource not found: " + request->name());
+    }
+
+    std::shared_ptr<Camera> camera = std::dynamic_pointer_cast<Camera>(rb);
+    std::shared_ptr<std::unordered_map<std::string, std::shared_ptr<ProtoType>>> command = struct_to_map(request->command());
+    std::shared_ptr<std::unordered_map<std::string, std::shared_ptr<ProtoType>>> result = camera->do_command(request->name(), command);
+
+    *response->mutable_result() = map_to_struct(result);
+
+     return ::grpc::Status(); 
+
+} 
+
+
+::grpc::Status CameraServer::GetImage(
+    ::grpc::ServerContext* context,
+    const ::viam::component::camera::v1::GetImageRequest* request,
+    ::viam::component::camera::v1::GetImageResponse* response) {
+
+    if (request == nullptr) {
+        return ::grpc::Status(::grpc::StatusCode::INVALID_ARGUMENT,
+                              "Called [GetImage] without a request");
+    };
+
+
+     std::shared_ptr<ResourceBase> rb = sub_svc->resource(request->name());
+    if (rb == nullptr) {
+        return grpc::Status(grpc::UNKNOWN, "resource not found: " + request->name());
+    } 
+
+   std::shared_ptr<Camera> camera =std::dynamic_pointer_cast<Camera>(rb);
+
+    std::string name = request->name();
+    std::string mime_type = request->mime_type();
+    Camera::raw_image image = camera->get_image(name, mime_type);
```

Comment:
(nit) these three lines can be combines. the type information of `name` and `mime_type` aren't particularly interesting, so just calling `camera->get_image(request->name(), request->mime_type())` should be sufficient :)